CIScripts/Test/PesterLogger/RemoteLogCollector.ps1

```diff
@@ -1,11 +1,94 @@
+. $PSScriptRoot/../../Common/Aliases.ps1
+. $PSScriptRoot/../../Common/Invoke-NativeCommand.ps1
+
 . $PSScriptRoot/PesterLogger.ps1
 
-function New-LogSource {
+class LogSource {
+    [System.Management.Automation.Runspaces.PSSession] $Session
+
+    [Hashtable] GetContent() {
+        throw "LogSource is an abstract class, use specific log source instead"
+    }
+
+    ClearContent() {}
+}
+
+class FileLogSource : LogSource {
+    [String] $Path
+
+    [Hashtable] GetContent() {
+        $ContentGetterBody = {
+            Param([Parameter(Mandatory = $true)] [string] $From)
+            $Files = Get-ChildItem -Path $From -ErrorAction SilentlyContinue
+            $Logs = @{}
+            if (-not $Files) {
+                $Logs[$From] = "<FILE NOT FOUND>"
+            } else {
+                foreach ($File in $Files) {
+                    $Content = Get-Content -Raw $File
+                    $Logs[$File.FullName] = if ($Content) {
+                        $Content
+                    } else {
+                        "<FILE WAS EMPTY>"
+                    }
+                }
+            }
+            return $Logs
+        }
+        return Invoke-CommandRemoteOrLocal -Func $ContentGetterBody -Session $this.Session -Arguments $this.Path
+    }
+
+    ClearContent() {
+        $LogCleanerBody = {
+            Param([Parameter(Mandatory = $true)] [string] $What)
+            $Files = Get-ChildItem -Path $What -ErrorAction SilentlyContinue
+            foreach ($File in $Files) {
+                try {
+                    Remove-Item $File
+                }
+                catch {
+                    Write-Warning "$File was not removed due to $_"
+                }
+            }
+        }
+        Invoke-CommandRemoteOrLocal -Func $LogCleanerBody -Session $this.Session -Arguments $this.Path
+    }
+}
+
+class ContainerLogSource : LogSource {
+    [String] $Container
+
+    [Hashtable] GetContent() {
+        $Command = Invoke-NativeCommand -Session $this.Session -CaptureOutput -AllowNonZero {
+            docker logs $Using:this.Container
+        }
+        return @{
+            "$( $this.Container ) container logs" = $Command.Output
+        }
+    }
+}
+
+function New-ContainerLogSource {
```

Comment:
It may actually return multiple elements. Function name may be misleading. The same remark applies to New-FileLogSource.

Reply:
This is a standard practice in powershell, for example, `New-Item` usually returns just one element, but
```powershell
New-Item -Type file -Path "foo", "bar", "baz"
```
creates multiple files.

There's even a lint in powershell analyzer that forbids plural nouns (but it's disabled in our CI)